tests/e2e/utils/src/pages/admin-edit.js

```diff
@@ -0,0 +1,55 @@
+import { waitForTimeout } from '../flows/utils';
+
+
+export class AdminEdit {
+	/**
+	 * Publish the object being edited and verify published status
+	 *
+	 * @param button Publish button selector
+	 * @param publishNotice Publish notice selector
+	 * @param publishVerification Expected notice on successful publish
+	 * @returns {Promise<void>}
+	 */
+	async verifyPublish( button, publishNotice, publishVerification ) {
+		// Wait for auto save
+		await waitForTimeout( 2000 );
+
+		// Publish and verify
+		await expect( page ).toClick( button );
+		await page.waitForSelector( publishNotice );
+		await expect( page ).toMatchElement( publishNotice, { text: publishVerification } );
+	}
+
+	/**
+	 * Get the ID of the object being edited
+	 *
+	 * @returns {Promise<*>}
+	 */
+	async getId() {
+		let postId = await page.$( '#post_ID' );
+		let objectID = await page.evaluate( element => element.value, postId );
+		return objectID;
+	}
+}
+
+export class OrderEdit extends AdminEdit {
+	/**
+	 * Publish the object being edited and verify published status
+	 *
+	 * @param button Publish button selector
+	 * @param publishNotice Publish notice selector
+	 * @param publishVerification Expected notice on successful publish
+	 * @returns {Promise<void>}
+	 */
+	async verifyPublish( button, publishNotice, publishVerification ) {
+		await super.verifyPublish( button, publishNotice, publishVerification );
+
+		await expect( page ).toMatchElement( '#select2-order_status-container', { text: 'Processing' } );
+		await expect( page ).toMatchElement(
+			'#woocommerce-order-notes .note_content',
+			{
+				text: 'Order status changed from Pending payment to Processing.',
```

Comment:
Do we always expect the `Processing` status and note status text to be present (e.g., `Order status changed from Pending payment to Processing.` ) when using `verifyPublish()` here, or do we need to make them more dynamic?

Reply:
@zhongruige Thanks for pointing this out. That test is specific to our test flow so I moved that test to there. That eliminated the need for the `OrderEdit` class so I removed it.

This is ready for another review.